cli/cli_app.go

```diff
@@ -264,6 +266,57 @@ func shouldPrintTerraformHelp(terragruntOptions *options.TerragruntOptions) bool
 	return false
 }
 
+// loadEnvVars loads environment variable definitions from a shell command.
+// Currently the function expects KEY=value output to STDOUT
+// the method returns the old evironment variables map (dictionary) if the
+// changes should be transient.
+func loadEnvVars(terragruntOptions *options.TerragruntOptions, curHook config.Hook) (map[string]string, error) {
```

Comment:
Having read through this whole method, it's trying to do many things, and may be hiding some bugs (see comments below for more context). I'd recommend breaking it down further:

1. One method that executes the LoadEnvVars hook and returns stdout as a string.
1. One method that, given a string, parses env vars from it, and returns those env vars as a map. 
1. One method that, given env vars as a map, returns a clone of `terragruntOptions` with the new env vars set.

You can then add unit tests for each of these items to make sure they work as intended.